### DISCUSSION TITLE
OneDrive Client Version v2.4.24 - Deprecation of sha1Hash on OneDrive Personal

### DISCUSSION THREAD

**Discussion by abraunegg** [abraunegg/onedrive#2430] [General]
Microsoft has announced the deprecation of sha1Hash on OneDrive Personal starting July 2023.

OneDrive Client Version v2.4.24 has been released to ensure that there are no impacts because of this announcement, however it is *critical* that you update to v2.4.24 as soon as possible to ensure you are not impacted by the older (v2.4.23 and below) which has to the potential to use sha1 hashes.

It will take a few days for package maintainers to update their packages - so please keep an eye out for new packages for your distribution, otherwise you will need to build from source.

**References:**
* https://devblogs.microsoft.com/microsoft365dev/deprecation-of-sha1hash-on-onedrive-personal/

**Comment by Rmano** [comment]
Hi @abraunegg --- thank you very much. I just upgraded --- and on restart, I had this warning:

     Jun 20 10:00:59 pern onedrive[296384]: The item database is incompatible, re-creating database table structures

and now the client is printing

     Jun 20 10:08:08 pern onedrive[296384]: ERROR: OneDrive API inconsistency - the file does not have any hash 

every 5-6 seconds. Is it a normal effect of the change of hashes? Or it is something I'm doing wrong? Should I do a --resync?

```
[romano:~/software/onedrive/onedrive] tag: v2.4.24* 1 ± onedrive --display-config
Configuration file successfully loaded
onedrive version                             = v2.4.24
Config path                                  = /home/romano/.config/onedrive
Config file found in config path             = true
Config option 'sync_dir'                     = /home/romano/OneDrive
Config option 'enable_logging'               = false
Config option 'log_dir'                      = /var/log/onedrive/
Config option 'disable_notifications'        = false
Config option 'min_notify_changes'           = 5
Config option 'skip_dir'                     = Attic|Notebooks|Backups|Blocs de Notas
Config option 'skip_dir_strict_match'        = false
Config option 'skip_file'                    = ~*|.~*|*.tmp|Romano @ Universidad*
Config option 'skip_dotfiles'                = false
Config option 'skip_symlinks'                = false
Config option 'monitor_interval'             = 60
Config option 'monitor_log_frequency'        = 6
Config option 'monitor_fullscan_frequency'   = 100
Config option 'read_only_auth_scope'         = false
Config option 'dry_run'                      = false
Config option 'upload_only'                  = false
Config option 'download_only'                = false
Config option 'local_first'                  = false
Config option 'check_nosync'                 = false
Config option 'check_nomount'                = false
Config option 'resync'                       = false
Config option 'resync_auth'                  = false
Config option 'cleanup_local_files'          = false
Config option 'classify_as_big_delete'       = 1000
Config option 'disable_upload_validation'    = false
Config option 'bypass_data_preservation'     = false
Config option 'no_remote_delete'             = false
Config option 'remove_source_files'          = false
Config option 'sync_dir_permissions'         = 700
Config option 'sync_file_permissions'        = 600
Config option 'space_reservation'            = 52428800
Config option 'application_id'               = 
Config option 'azure_ad_endpoint'            = 
Config option 'azure_tenant_id'              = common
Config option 'user_agent'                   = 
Config option 'force_http_11'                = false
Config option 'debug_https'                  = false
Config option 'rate_limit'                   = 0
Config option 'operation_timeout'            = 3600
Config option 'dns_timeout'                  = 60
Config option 'connect_timeout'              = 10
Config option 'data_timeout'                 = 600
Config option 'ip_protocol_version'          = 0
Config option 'sync_root_files'              = false
Selective sync 'sync_list' configured        = false
Config option 'sync_business_shared_folders' = true
Business Shared Folders configured           = true
business_shared_folders contents:
#
# después de modificar hacer
# onedrive --resync
# 
# Test por si acaso
# Bioingeniería
#

Config option 'webhook_enabled'              = false
```

I rolled back to 2.4.23 for now... but I can check and help testing if you need.

**Comment by Rmano** [comment]
I tried a `--resync`, still the same error.

**Comment by abraunegg** [comment]
@Rmano 
The warning about being incompaible is normal - as this is a DB change.

This message:
```
ERROR: OneDrive API inconsistency - the file does not have any hash
```

Is also normal ... there is no hash for that file .. or at least not what is being expected.

To diagnose this further, please open a bug report, and please generate a verbose debug log and email this ASAP - as this will be the only way to identify what is going on for you.

**Comment by Rmano** [comment]
Ok, I'll try. Maybe I just have to wait until the first scan is finished? (I have a very big, ignored directory that nevertheless slows things down). I'll try to dig how to do the verbose log (I know, you explained that to me last time... but... 😳)

**Reply by abraunegg** [reply]
If in doubt - follow: https://github.com/abraunegg/onedrive/wiki/Generate-debug-log-for-support

What 'account type' is this as well?

There is certainly something odd going on with your data online ... and I cant say what without a debug log

**Reply by Rmano** [reply]
I am generating one now.  The account type is... well, I am not sure. This is a Univeristy account with a full Office365/OneDrive account, I suppose it's professional. But MS is doing such a mess between shared folder, share point, teams folders, whatever, that I lost them, to be honest.

I will email you a partial log, I have already the error in it. I will let it go (I am running 

```
[romano:~/software/onedrive] % onedrive --display-config > debug_output.log 2>&1
[romano:~/software/onedrive] % echo >> debug_output.log 
[romano:~/software/onedrive] % onedrive --monitor --verbose --verbose >> debug_output.log 2>&1
```

...no really sure when to stop it ;-).

**Reply by abraunegg** [reply]
If your log has already captured it - then yes you can stop and email please

**Reply by Rmano** [reply]
Sent!

**Comment by abraunegg** [comment]
@Rmano 
Many thanks for the partial log file.

I have analysed the file, and the data, as provided by OneDrive does not contain a hash - which is why the error message is being presented - so this is all 100% accurate.

As per my email response, please raise a case with Microsoft to solve your issue.

**Comment by Rmano** [comment]
Another question: I added this:

```
diff --git a/src/sync.d b/src/sync.d
index 663a0de..5db6ec8 100644
--- a/src/sync.d
+++ b/src/sync.d
@@ -188,7 +188,9 @@ private Item makeItem(const ref JSONValue driveItem)
                // No hashes ..
                if ((item.quickXorHash.empty) && (item.sha256Hash.empty) ) {
                        // Odd .. no hash ......
-                       log.error("ERROR: OneDrive API inconsistency - the file does not have any hash");
+                       /* string path = "path" in driveItem ? driveItem["path"].str : null; */
+                       log.error("ERROR: OneDrive API inconsistency - the file does not have any hash");
+                       log.error("CONTD: file name: " ~ item.name);
                }
        }       
```
to print the name of the problematic file (s), but even if I tried a bit of blind shots I was not able to print the full path of them... ¿is it possible? 

Thanks!

(Notice that I know nothing of "D", so this is "by feel" and can be horribly wrong).

**Reply by abraunegg** [reply]
Not directly .. the path needs to be calculated.

See https://github.com/abraunegg/onedrive/pull/2436/files where this is calculated and https://github.com/abraunegg/onedrive/issues/2433#issuecomment-1599525768 for instructions on how to use the PR
